### ISSUE TITLE
Hover observables are never unsubscribed

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#1190]
Repro:

Create a hover provider in an extension like this:

```typescript
    sourcegraph.languages.registerHoverProvider(['*'], {
        provideHover: () =>
            interval(100).pipe(
                map(n => ({
                    contents: { kind: sourcegraph.MarkupKind.Markdown, value: `⏱ ️${n}` },
                }))
            ),
    })
```

1. Enable the extension.
2. Turn on extension host communication trace logging by running `localStorage.debug=1;location.reload()` and then toggling it on in the `Ext` menu in the bottom right.
3. Hover over it in Sourcegraph. You will see the hover change frequently.
4. Now hover away from it.

Notice that the JS devtools console shows that it is still sending messages with new hover contents. It should have been unsubscribed.

This is a resource leak and could make things slow. Not high priority since it's unlikely to cause many actual problems, but good to fix anyway.

**Comment by felixfbecker** [comment]
Also seems to happen for references providers after the panel already got closed. TypeScript will keep searching dependent after dependent (and show progress indicators)

**Comment by felixfbecker** [comment]
Reopening this because this test still fails consistently:

```ts
    test('abort request with observable emission', async () => {
        const [serverTransports, clientTransports] = createMessageTransports()
        const server = createConnection(serverTransports)
        const { wait, done } = createBarrier()
        server.onRequest('m', params => new Observable<never>(observer => done))
        server.listen()
        const client = createConnection(clientTransports)
        client.listen()
        const subscription = client.observeRequest('m').subscribe()
        subscription.unsubscribe()
        await wait
    })
```

**Comment by lguychard** [comment]
Closed by #1768
